nexus/src/sagas.rs

```diff
@@ -223,3 +246,190 @@ async fn sic_instance_ensure(
         .map(|_| ())
         .map_err(ActionError::action_failed)
 }
+
+/*
+ * "Migrate Instance" saga template
+ */
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ParamsInstanceMigrate {
+    // TODO design: We can't just use db::model::Instance here as it doesn't impl
+    // Serialize so instead we need the project_id/instance_name to look it up
+    // in the saga.
```

Comment:
It's not exactly an issue, more of an observation. We already look up the Instance before executing the saga so it would be nice to just pass it along to the saga, but the PR is big enough as is 😅️

Reply:
I think this is going to be a common pattern.  There are tradeoffs for both serializing the whole row and just parts of it.  (The big one I'm thinking of is that if we serialize the whole row, then it seems like that's going to make schema migrations even more complicated because even once we've migrated the database schema and even if we do subsequent Nexus upgrades, Nexus still needs to know about old schema versions in case it recovers a saga with an older version of the row baked into its persistent state)   I don't think we have enough examples yet to know how best to deal with this.

Reply:
Yea, for now I'll leave this as-is just doing the lookup in the saga instead of serializing the Instance model.